src/enclave/enclave_event_channel.c

```diff
@@ -172,14 +179,15 @@ void initialize_enclave_event_channel(
 
         *dev_id = enc_dev_config[i].dev_id;
 
-        struct lthread* lt = NULL;
         if (lthread_create(
-                &lt, NULL, vio_enclave_process_host_event, (void*)dev_id) != 0)
+                &vio_tasks[i],
```

Comment:
Does that actually solve the underlying problem? `lthread_create` follows the same pattern (`*new_lt` vs `lt`), without locking as far as I can tell.

Reply:
I think that it should be fine. The race only happens when `vio_enclave_process_host_event()` starts running, which only happens after lthread_create() has updated `*new_lt`.

Reply:
Yeah, you're right, that looks fine!